### ISSUE TITLE
How to generate input data with proper shapes

### ISSUE THREAD

**Issue by orenpapers** [philipperemy/cond_rnn#15]
I have a list of 564 time series. Each time series is 1d vector of different sizes (e.g. 366, 558, 812, ...).
I also have a respective dataframe of 564X3 that holds the condition (categorical features).
My batch size is 4 and timestamps is 10.
What is the best way to transform these data to the proper data to use conditional RNN?
(3-D Tensor with shape [batch_size, timesteps, input_dim] from series list of timeseries, and 2-D Tensor or list of tensors with shape [batch_size, cond_dim])?

**Comment by philipperemy** [comment]
@hstarmans I posted an example here: https://github.com/philipperemy/cond_rnn/blob/master/examples/temp.py.
Let me know if it solves your problem. Each time series has a different size in the dataset.

**Comment by orenpapers** [comment]
@philipperemy Thanks, I see that the shape of x is `(200000, 100, 1)`.
Can you please explain what does these numbers represent?

**Comment by philipperemy** [comment]
```
(200000, 100, 1)
```
First is the batch dim, second is the time dim, and last is the input dim.

Refer to this documentation: https://keras.io/api/layers/recurrent_layers/simple_rnn/

**Comment by orenpapers** [comment]
@philipperemy Thanks, by batch dim you mean number of samples in every batch?

**Comment by philipperemy** [comment]
Yeah, correct. The dimension that you can parallelize on.

**Comment by orenpapers** [comment]
@philipperemy Is it chosen arbitrary? I mean, could u just set it to 100000 or is it dictated by other hyper-parameters / dimensions?

**Comment by philipperemy** [comment]
Nope purely arbitrary.
